include/reactor-cpp/port.hh

```diff
@@ -24,36 +26,89 @@ private:
   const PortType type_;
 
+  multiport::LockedPortList active_ports_{};
+  std::size_t index_ = 0;
+
   std::set<Reaction*> dependencies_{};
   std::set<Reaction*> triggers_{};
-  std::set<Reaction*> antidependencies_{};
+  std::set<Reaction*> anti_dependencies_{};
 
 protected:
+  bool present_{false}; // NOLINT cppcoreguidelines-non-private-member-variables-in-classes
+
   BasePort(const std::string& name, PortType type, Reactor* container)
       : ReactorElement(name, (type == PortType::Input) ? ReactorElement::Type::Input : ReactorElement::Type::Output,
                        container)
       , type_(type) {}
 
+  BasePort(const std::string& name, PortType type, Reactor* container, multiport::LockedPortList active_ports,
+           std::size_t index)
+      : ReactorElement(name, (type == PortType::Input) ? ReactorElement::Type::Input : ReactorElement::Type::Output,
+                       container)
+      , type_(type)
+      , active_ports_(active_ports)
+      , index_(index) {}
   void base_bind_to(BasePort* port);
-  void register_dependency(Reaction* reaction, bool is_trigger);
-  void register_antidependency(Reaction* reaction);
+  void register_dependency(Reaction* reaction, bool is_trigger) noexcept;
+  void register_antidependency(Reaction* reaction) noexcept;
   virtual void cleanup() = 0;
 
 public:
   [[nodiscard]] inline auto is_input() const noexcept -> bool { return type_ == PortType::Input; }
   [[nodiscard]] inline auto is_output() const noexcept -> bool { return type_ == PortType::Output; }
+  [[nodiscard]] inline auto is_present() const noexcept -> bool {
+    if (has_inward_binding()) {
+      return inward_binding()->is_present();
+    }
+    return present_;
+  };
 
   [[nodiscard]] inline auto has_inward_binding() const noexcept -> bool { return inward_binding_ != nullptr; }
   [[nodiscard]] inline auto has_outward_bindings() const noexcept -> bool { return !outward_bindings_.empty(); }
   [[nodiscard]] inline auto has_dependencies() const noexcept -> bool { return !dependencies_.empty(); }
-  [[nodiscard]] inline auto has_antidependencies() const noexcept -> bool { return !antidependencies_.empty(); }
+  [[nodiscard]] inline auto has_anti_dependencies() const noexcept -> bool { return !anti_dependencies_.empty(); }
 
   [[nodiscard]] inline auto inward_binding() const noexcept -> BasePort* { return inward_binding_; }
   [[nodiscard]] inline auto outward_bindings() const noexcept -> const auto& { return outward_bindings_; }
 
   [[nodiscard]] inline auto triggers() const noexcept -> const auto& { return triggers_; }
   [[nodiscard]] inline auto dependencies() const noexcept -> const auto& { return dependencies_; }
-  [[nodiscard]] inline auto antidependencies() const noexcept -> const auto& { return antidependencies_; }
+  [[nodiscard]] inline auto anti_dependencies() const noexcept -> const auto& { return anti_dependencies_; }
+
+  [[nodiscard]] inline auto activate() const -> bool {
```

Comment:
I am pretty confused about the `activate()` `clear()` and `deactivate()` methods. Could you explain what they do and maybe also add a bit of documentation?

Reply:
I think that my confusion mostly came from terminology. In Lf we say "present" and absent, not "activated" and "deactivated". Could you update the variable and function names? 

I still don't understand `clear()` means in comparison to `deactivate` though.

Reply:
changed